### PR TITLE
feat: Introduce Passphrase Cache and other features

### LINKED ISSUE #25: Use secret storage API to remember passphrase
VS Code provides secret storage API for managing sensitive data across workspaces.

See <https://code.visualstudio.com/api/references/vscode-api#SecretStorage> for more information.

We may use this API to remember passphrases for user in a safe way.

### LINKED ISSUE #40: Reload key when git config singingKey has changed
When editing the git.user.singingKey manually, it still uses the old one until reload. Would be nice if it would use the updated one.

### PR DESCRIPTION
**Pull request by AnnAngela**
My local repo had been messed up by myself so I have to delete it and clone back again, which caused #48 been closed, sorry for that. I need to learn git more deeply :(

See #48
Resolve #25
Resolve #40

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -18,4 +18,4 @@
         "node_modules",
         ".vscode-test"
     ]
-}
```
Comment (wdhongtw) [inline tsconfig.json:21]: EOL here. :D

File: src/logger.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     * Log some message at error level.
+     * @param message - a message without ending new line
+     */
+    error(message: string): void {
+        if (this.level < LogLevel.error) {
+            return;
+        }
+        this.outputChannel.appendLine(`[${toLocalTimeZoneString()}] [ERROR] ` + message);
+    }
+}
```
Comment (wdhongtw) [inline src/logger.ts:71]: EOL here. :D

**Review by wdhongtw — COMMENTED** [review]
It's ok. :D

Just two file need EOL at last line, and we can merge this.

1. add EOL
2. `git add <files>`
3. `git commit --amend`
4. push again

You can do it well this time. :D

**Comment by AnnAngela** [comment]
@wdhongtw Done. I forgot to change my vscode setting, sorry for that.

**Pull request merged**